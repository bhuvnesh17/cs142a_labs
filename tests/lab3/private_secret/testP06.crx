// test mutual recursion

func isOdd(a:int) : int {
  if (0 == a) {
    return false;
  }
  return ::isEven(a-1);
}

func isEven(a:int) : int {
  if (0 == a) {
    return true;
  }
  return ::isOdd(a-1);
}
