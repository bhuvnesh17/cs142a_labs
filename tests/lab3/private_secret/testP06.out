Error parsing file.
ResolveSymbolError(7,12)[Could not find isEven.]
Symbol(readInt)
Symbol(readFloat)
Symbol(printBool)
Symbol(printInt)
Symbol(printFloat)
Symbol(println)
Symbol(isOdd)
  Symbol(a)


